### ISSUE TITLE
Const type serialization not implemented yet: CONST Char type=kotlin.Char value=

### ISSUE THREAD

**Issue by soywiz** [JetBrains/kotlin-native#1717]
```
exception: kotlin.NotImplementedError: An operation is not implemented: Const type serialization not implemented yet: CONST Char type=kotlin.Char value=
        at org.jetbrains.kotlin.backend.konan.serialization.IrSerializer.serializeConst(SerializeIr.kt:240)
        at org.jetbrains.kotlin.backend.konan.serialization.IrSerializer.serializeExpression(SerializeIr.kt:495)
        at org.jetbrains.kotlin.backend.konan.serialization.IrSerializer.serializeIrFunction(SerializeIr.kt:560)
        at org.jetbrains.kotlin.backend.konan.serialization.IrSerializer.serializeDeclaration(SerializeIr.kt:650)
        at org.jetbrains.kotlin.backend.konan.serialization.IrSerializer.encodeDeclaration(SerializeIr.kt:695)
        at org.jetbrains.kotlin.backend.konan.serialization.IrSerializer.serializeInlineBody(SerializeIr.kt:77)

```

From: https://github.com/korlibs/korlibs/blob/a9fd6f865dd7f6274a654a3d78659d6c5fda058f/kds/kds/common/src/com/soywiz/kds/Array2.kt#L20

```
inline operator fun <reified T> invoke(
			map: String,
			//marginChar: Char = '\u0000', // <---- produces the problem
			marginChar: Char = 0.toChar(), // <---- workaround
			gen: (char: Char, x: Int, y: Int) -> T
		): Array2<T> {
```

**Comment by soywiz** [comment]
This was reproduced in 0.7, but can't reproduce on `0.8-dev-2494`

**Comment by soywiz** [comment]
Sorry: Reproduced in `0.8-dev-2494 ` too. It seems that something was cached, but building from a clean state reproduces the issue:

```
 * Source files: Array2.kt, BitSet.kt, CacheMap.kt, CircularList.kt, Computed.kt, DoubleArrayList.kt, DoubleCircularList.kt, DoubleLinkedList.kt, DoubleQueue.kt, DoubleStack.kt, Extra.kt, FastIntMap.kt, IntArrayList.kt, IntCircularList.kt, IntExt.kt, IntFloatMap.kt, IntIntMap.kt, IntLinkedList.kt, IntMap.kt, IntQueue.kt, IntSet.kt, IntStack.kt, LinkedList.kt, ListExt.kt, ListReader.kt, MapExt.kt, MapList.kt, MemTools.kt, MutableListExt.kt, Pool.kt, PriorityQueue.kt, Queue.kt, Stack.kt, WeakMap.kt, WeakProperty.kt, binarySearch.kt, ds.kt, mapWhile.kt, FastIntMapNative.kt, WeakMapNative.kt
 * Compiler version info: Konan: 0.8-dev-2494 / Kotlin: 1.2.60
 * Output kind: LIBRARY

exception: kotlin.NotImplementedError: An operation is not implemented: Const type serialization not implemented yet: CONST Char type=kotlin.Char value=
        at org.jetbrains.kotlin.backend.konan.serialization.IrSerializer.serializeConst(SerializeIr.kt:240)
```

**Comment by alexander-gorshenev** [comment]
@soywiz Should have been fixed by #1729
